dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/instrumentation/network/HttpSpecRequestInfo.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+package com.datadog.android.api.instrumentation.network
+
+import java.io.IOException
+
+/**
+ * Represents information about an HTTP request (only that is specified with the HTTP specification).
+ * Isolates specific library's details from the Datadog SDK.
+ */
+interface HttpSpecRequestInfo {
+    /**
+     * The URL associated with the HTTP request.
+     */
+    val url: String
+
+    /**
+     * Represents the HTTP headers associated with the request.
+     */
+    val headers: Map<String, List<String>>
+
+    /**
+     * The MIME type of the payload associated with the HTTP request or response, represented
+     * as a string. Can be null if the content type is unspecified.
+     *
+     * The existence of this value depends on the specific implementation and may be null.
```

Comment:
If we are talking about HTTP, probably all implementations of HTTP transport engine support it, it is just the client/server who can omit it, since this header is completely optional.